custom/icds/data_management/tasks.py

```diff
@@ -1,11 +1,36 @@
+import io
+
+from django.template.loader import render_to_string
+
 from celery.task import task
 
+from corehq.util.log import send_HTML_email
 from custom.icds.data_management.models import DataManagementRequest
+from custom.icds.data_management.serializers import (
+    DataManagementRequestSerializer,
+)
 
 
 @task(queue='background_queue')
-def execute_data_management(slug, domain, db_alias, initiated_by, start_date, end_date):
-    processed, skipped = DataManagementRequest(
-        slug=slug, domain=domain, db_alias=db_alias, initiated_by=initiated_by,
-        start_date=start_date, end_date=end_date
-    ).execute()
+def execute_data_management(request_id):
+    request = DataManagementRequest.objects.get(pk=request_id)
+    processed, skipped, logs = request.execute()
+    serialized = DataManagementRequestSerializer().to_representation(request)
```

Comment:
it appears this is only used for template rendering. In that case it doesn't seem necessary to have a serializer at all and you can just reference the data model in the template directly. You can do date formatting with the [date](https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#date) template filter.

Reply:
we are fetching the objects via AJAX and then rendered via KnockoutJS. So i guess can't use django template filters.

Reply:
I don't follow, this serialised data is passed to `render_to_string` which renders the email body from a Django template. As far as I can tell that's the only place you use this serialiser.

Reply:
it's used in `paginate_data_management_requests` as well.